.kokoro/downstream-client-library-check.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Presubmit to ensure the dependencies of the Google Libraries BOM, with the modification of change
+# in the PR, pick up the highest versions among transitive dependencies.
+# https://maven.apache.org/enforcer/enforcer-rules/requireUpperBoundDeps.html
+
+set -eo pipefail
+# Display commands being run.
+set -x
+
+REPO=$1
+## Get the directory of the build script
+scriptDir=$(realpath $(dirname "${BASH_SOURCE[0]}"))
+## cd to the parent directory, i.e. the root of the git repo
+cd ${scriptDir}/..
+
+# Make java-core artifacts available for 'mvn validate' at the bottom
+mvn install -DskipTests=true -Dmaven.javadoc.skip=true -Dgcloud.download.skip=true -B -V -q
+
+# Read the current version of this java-core in the POM. Example version: '0.116.1-alpha-SNAPSHOT'
+AUTH_VERSION_POM=pom.xml
+# Namespace (xmlns) prevents xmllint from specifying tag names in XPath
+AUTH_VERSION=`sed -e 's/xmlns=".*"//' ${AUTH_VERSION_POM} | xmllint --xpath '/project/version/text()' -`
```

Comment:
As of now, I see this script is specific to auth library (from the variable name "AUTH_VERSION)). But I believe this script is beneficial to Google HTTP client, GAX, and all other libraries in the shared dependencies BOM. What do you think about making it generic?

Reply:
I will update it to be more generic.